src/database/database-internal.ts

```diff
@@ -96,9 +106,43 @@ export class DatabaseService {
 
       this.databases[dbUrl] = db;
     }
+
+    if (!this.tokenRefresh) {
+      this.tokenRefresh = true;
```

Comment:
Optional suggestion:

You can make this a little simpler if you combine `tokenRefresh` and `tokenRefreshTimeout`. You could initialize `tokenRefreshTimeout` to `undefined` initially and then set the schedule right here, which seems more intuitive to me. With the current setup, I have to look at `onTokenChange` to figure out the scheduling.

I also wonder if we should use `setInterval`. It seems a little more appropriate, but might make things more complicated in some places.

Reply:
This code is just intended to register the token change listener. There's no guarantee that the listener will fire immediately. So theoretically it is possible for `tokenRefresh` to be true, while `tokenRefreshTimeout` is not yet set. What we really need is a way to track whether we have registered the listener or not (so we can avoid duplicate registrations, and also unregister it at delete). So I'm renaming `tokenRefresh` to `tokenListenerRegistered`. That should clear up any confusions, and also address the above comment regarding naming.

As for `setInterval()`, given that we can receive tokens with arbitrary expiration times from auth (we've seen tokens with expiry times like 30min and 45min), an interval scheduler is not very useful. So we use the token listener to watch for token changes, and schedule refresh events according to the expiry time we get.